inference-engine/tests/functional/plugin/plaidml/shared_tests_instances/single_layer_tests/embedding_bag_offsets_sum.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vector>
+
+#include "single_layer_tests/embedding_bag_offsets_sum.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+
+namespace {
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::I32,
+};
+
+const std::vector<InferenceEngine::Precision> indPrecisions = {
+        InferenceEngine::Precision::I32
+};
+
+const std::vector<std::vector<size_t>> embTableShape = {{5, 6}, {10, 35}, {5, 4, 16}};
+const std::vector<std::vector<size_t>> indices =
+        {{0, 1, 2, 2, 3}, {4, 4, 3, 1, 0}, {1, 2, 1, 2, 1, 2, 1, 2, 1, 2}};
+const std::vector<std::vector<size_t>> offsets = {{0, 2}, {0, 0, 2, 2}, {2, 4}};
+const std::vector<size_t> defaultIndex = {0, 4};
+const std::vector<bool> withWeights = {false};
+const std::vector<bool> withDefaultIndex = {true};
```

Comment:
Similar to the one mentioned above. [EmbeddingBagOffsetsSumLayerTest::SetUp](https://github.com/Flex-plaidml-team/openvino/blob/zhibinli-embedding_bag_offsets_sum/inference-engine/tests/functional/plugin/shared/src/single_layer_tests/embedding_bag_offsets_sum.cpp) always reads in `defaultIndex ` even if `withDefaultIndex = false`. Is there any limitation of the ordering of `ctx.operands.at(N)` by the way?

Reply:
The operands have a required order, that is supposed to match the order of "Inputs" in the documentation.

It appears that when `withDefaultIndex` is `false`, the `defaultIndex` parameter is not used even though it is passed to `makeEmbeddingBagOffsetsSum`; in `inference-engine/tests/ngraph_functions/src/embedding_bag_offsets_sum.cpp` notice how the shorter constructor for `opset3::EmbeddingBagOffsetsSum` is used in this case.

So I think when `withDefaultIndex = false` the value of `defaultIndex` is correctly ignored.

Reply:
I see and thanks for the info about the setup process! It looked a little bit confusing to me at the very beginning. 

Just one more question. Why in the tests `defaultIndex` is always provided while `per_sample_weights` is always NOT? Isn't that a inconsistency or is there any other extra consideration?